### ISSUE TITLE
Improve table printing

### ISSUE THREAD

**Issue by DominiqueMakowski** [easystats/report#4]
Table printing is a bit jiggly, i.e., the columns are not perfectly aligned (due to coloured columns). This must be improved, but I am not sure how...

The answer is probably within the `.display.data.frame` function in `display.R`, where the alignment is done.

**Comment by strengejacke** [comment]
I digged a bit into the code, and this one is tricky... we need to use `format()` on the complete column, e.g. in `.colour_column_if()`. But once it's a string, we can't check for the negative / positive sign... ?!?

**Comment by DominiqueMakowski** [comment]
Yes, I tried with some regex detection (if the stripped characters began with the "-" string), without success... 

We might need to imagine another way of integrating the colouring, the formatting and the to-string conversion than my clunky proposal

**Comment by DominiqueMakowski** [comment]
Moreover, the format has to be used on the striped columns (without the colour codes), otherwise, the columns are too large (as the colour codes disappear)... So yea very tricky, puzzling and challenging haha... and worth trouble (even now with the uneven columns, I find the colouring of the coefs and such super cool 😄)

**Comment by strengejacke** [comment]
Ok, I think we can reduce code complexity and should put the formatting in `.colour_column_if()`. Now we have to get rid of `NA` where it makes no sense.

**Comment by strengejacke** [comment]
![unbenannt](https://user-images.githubusercontent.com/26301769/52913293-db89ea80-32bc-11e9-97a1-81a00003463c.png)

**Comment by DominiqueMakowski** [comment]
This can be relevant for future vignette / readme improvements: https://twitter.com/BrodieGaslam/status/1029059002225721344

**Comment by strengejacke** [comment]
I would say that column with numers should be right-aligned, while "string" columns (with values that are non-numeric from the user's perspective) should be left-aligned... Isn't this a common table convention?

**Comment by strengejacke** [comment]
Table printing no longer works for model-objects when digits are specified. Thus, we need some extra handling here. I tried to solve this using an additional class attribute. We should think about how to solve this issue, respectively if my suggestion is ok and not too sloppy.

**Comment by DominiqueMakowski** [comment]
![image](https://user-images.githubusercontent.com/8875533/52921688-c6f93100-3354-11e9-8150-518ac7822178.png)

It looks great 🎊 🎉 ! Brilliant implementation, thanks :)

**Comment by DominiqueMakowski** [comment]
It seems that *bold* slightly enlarges the column, resulting in a minor misalignment toward the end... but that's details ^^

**Comment by strengejacke** [comment]
Damn... ;-)
I have a fix in mind, will try this evening. We could, for the p-value column, replace all empty values with as many whitespaces as "column-width" (`nchar()`) and sourround the whitespaces with bold as well... Or we switch to a colour. But misalignment doesn't look that nice. :-)

**Comment by strengejacke** [comment]
Can you check if my workaround works for you? I don't have this alignment issue.

**Comment by DominiqueMakowski** [comment]
![image](https://user-images.githubusercontent.com/8875533/52984481-9f749800-342a-11e9-8398-37c998d1dcdc.png)

Seems like it's still there... are you using R studio also?

**Comment by strengejacke** [comment]
Yes, I'm using RStudio (both latest stable and current preview) on Windows 10 (light theme). Strange, but then let's format p-values in colour, not bold.

**Comment by DominiqueMakowski** [comment]
Strange indeed... maybe has something to do with the font used?
Will change back to yellow then (or maybe orange if yellow is too hard to read on a light theme?)
